pkg/applicationserver/io/grpc/grpc.go

```diff
@@ -150,11 +169,23 @@ func (s *impl) GetMQTTConnectionInfo(ctx context.Context, ids *ttnpb.Application
 	}, nil
 }
 
+var errNilApplicationUp = errors.DefineInvalidArgument("nil_application_up", "application upstream message is nil")
+
 func (s *impl) SimulateUplink(ctx context.Context, up *ttnpb.ApplicationUp) (*pbtypes.Empty, error) {
+	if up == nil {
+		return nil, errNilApplicationUp.New()
+	}
```

Comment:
Can't we catch this in validation?

Reply:
This is the top level gRPC request, which I'm not sure if it can be `nil` via the normal gRPC calls. I don't see in `applicationserver.*.pb.go` any way to have a `nil` request.

Reply:
Yeah indeed